scheduler/scheduler.go

```diff
@@ -43,22 +41,15 @@ type Scheduler struct {
 func (s *Scheduler) PrioritizeCandidates(resourceRequests []ResourceRequest, context Context) ([]string, error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-
-	hosts := filter(s.hosts, resourceRequests)
-
-	if len(hosts) == 0 {
-		return []string{}, nil
+	filteredHosts := []string{}
+	for host := range s.hosts {
+		filteredHosts = append(filteredHosts, host)
 	}
 
-	hs := hostSorter{
-		hosts:            hosts,
-		resourceRequests: resourceRequests,
+	filters := ProvideFilter()
```

Comment:
You basically hide/obfuscate the sort function by putting it inside the compute filter.
We should keep sorting as a separate call here to make it clear that that is happening as a seprate and distinct step from filtering out hosts